src/view/dataElements/xdmObject/components/booleanEdit.jsx

```diff
@@ -75,96 +44,67 @@ const BooleanInput = props => {
   // component a little more. Without adding more state management in this
   // component, it also means that when users switch input methods, the field's
   // value gets reset each time.
-  const [inputMethod, setInputType] = useState();
+  const [inputMethod, setInputMethod] = useState();
 
   useEffect(() => {
     const defaultInputMethod =
       typeof value === "boolean"
         ? inputMethods.CONSTANT
         : inputMethods.DATA_ELEMENT;
-    setInputType(defaultInputMethod);
+    setInputMethod(defaultInputMethod);
   }, [fieldName]);
 
   return (
-    <div>
-      <FieldLabel labelFor="inputMethodField" label="Input Method" />
-      <RadioGroup
-        id="inputMethodField"
-        selectedValue={inputMethod}
+    <Form>
+      <ReactSpectrumRadioGroup
+        label="Input Method"
+        orientation="horizontal"
+        value={inputMethod}
         onChange={newInputMethod => {
-          onChange("");
-          setInputType(newInputMethod);
+          setValue("");
+          setInputMethod(newInputMethod);
         }}
       >
         <Radio
           data-test-id="dataElementInputMethodField"
-          label="Provide Data Element"
           value={inputMethods.DATA_ELEMENT}
-        />
+        >
+          Provide Data Element
+        </Radio>
         <Radio
           data-test-id="valueInputMethodField"
-          className="u-gapLeft2x"
-          label="Select Value"
           value={inputMethods.CONSTANT}
-        />
-      </RadioGroup>
+        >
+          Select Value
+        </Radio>
+      </ReactSpectrumRadioGroup>
       {inputMethod === inputMethods.CONSTANT ? (
-        <div>
-          <FieldLabel labelFor="constantValueField" label="Value" />
-          <Select
-            data-test-id="constantValueField"
-            id="constantValueField"
-            value={convertActualValueToBooleanOptionValue(value)}
-            options={booleanOptions}
-            onChange={newValue => {
-              onChange(convertBooleanOptionValueToActualValue(newValue));
-            }}
-          />
-        </div>
+        <FormikRadioGroup
+          label="Value"
+          name={valueFieldName}
+          orientation="horizontal"
+        >
+          <Radio data-test-id="constantNoValueField" value="">
+            No Value
+          </Radio>
+          <Radio data-test-id="constantTrueField" value>
+            True
+          </Radio>
+          <Radio data-test-id="constantFalseField" value={false}>
+            False
+          </Radio>
```

Comment:
I switched from a Select to Radios because (1) it's what we should be using according to [spectrum guidelines](https://spectrum.adobe.com/page/combo-box/#Combo-box,-picker,-or-radio-buttons?) (2) Select/Picker doesn't support boolean values (mostly because they're used for React keys which must be strings) so you have to use strings instead and have a translation layer to get a boolean into formik state (see `convertActualValueToBooleanOptionValue` and `convertBooleanOptionValueToActualValue` in the old code for what that looks like).